pkg/providers/common/common.go

```diff
@@ -2,11 +2,105 @@ package common
 
 import (
 	_ "embed"
+	"fmt"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/bootstrapper"
+	"github.com/aws/eks-anywhere/pkg/crypto"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/types"
 )
 
 //go:embed config/audit-policy.yaml
 var auditPolicy string
 
+var NoProxyDefaults = []string{
+	"localhost",
+	"127.0.0.1",
+	".svc",
+}
+
+const (
+	privateKeyFileName = "eks-a-id_rsa"
+	publicKeyFileName  = "eks-a-id_rsa.pub"
+)
+
 func GetAuditPolicy() string {
 	return auditPolicy
 }
+
+func BootstrapClusterOpts(serverEndpoint string, clusterConfig *v1alpha1.Cluster) ([]bootstrapper.BootstrapClusterOption, error) {
+	env := map[string]string{}
+	if clusterConfig.Spec.ProxyConfiguration != nil {
+		noProxy := fmt.Sprintf("%s,%s", clusterConfig.Spec.ControlPlaneConfiguration.Endpoint.Host, serverEndpoint)
+		for _, s := range clusterConfig.Spec.ProxyConfiguration.NoProxy {
+			if s != "" {
+				noProxy += "," + s
+			}
+		}
+		env["HTTP_PROXY"] = clusterConfig.Spec.ProxyConfiguration.HttpProxy
+		env["HTTPS_PROXY"] = clusterConfig.Spec.ProxyConfiguration.HttpsProxy
+		env["NO_PROXY"] = noProxy
+	}
+	return []bootstrapper.BootstrapClusterOption{bootstrapper.WithEnv(env)}, nil
+}
+
+func ParseSSHAuthKey(key *string) error {
+	if len(*key) > 0 {
+		// When public key is entered by user in provider config, it may contain email address (or any other comment) at the end. ssh-keygen allows users to add comments as suffixes to public key in
+		// public key file. When CLI generates the key pair, no comments will be present. So we get rid of the comment from the public key to ensure unit tests that do string compare on the sshAuthorizedKey
+		// will pass
+		parts := strings.Fields(strings.TrimSpace(*key))
```

Comment:
I know you got this from vSphere but its pretty convoluted and could be much simpler. 

The function name is unrepresentative of whats happening. The function is  normalizing keys for our use-case by stripping the comment from a valid ssh authorized key as opposed to parsing an ssh auth key. It also returns error messages that make assumptions about the context under which it was called. It also has a subtle, and in my view hidden, behavior in that if the string is len(0) the call is a noop. That's cunning and code shouldn't be cunning.

```
func StripSshAuthorizedKeyComment(key string) (string, error) { 
  public, _, _, _, err := ssh.ParseAuthorizedKey([]byte(*key))
  if err != nil {
    return err
  }
  return string(ssh.MarshalAuthorizedKey(public))
}
```

Reply:
Will implement this function instead and remove the hidden behavior, requiring the caller to check whether to strip or not (depending on key presence). Also will remove the default trailing newline added by ssh.MarshalAuthorizedKey